### PR TITLE
a11y-tests: Direct a11y testing with SSR

### PR DESCRIPTION
**Pull request by statm**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

With this change, we will be able to perform a11y tests without having to start a test server. Instead of pointing puppeteer to test site URL (either localhost or Azure), we use react-dom to server side render interested components into a mini-html string, and feed it to puppeteer and axe-core.

Besides apparent performance/stability gain from not needing a server, with this approach we have complete control of test components (although for now they are still from `examples` folder), and their rendering process so utilities like `resetIds()` and `Stylesheet.reset()` can be used to reduce ID-related churn. Also, in the case of DetailsList and the likes, we can take advantage of Jest's mocking capability to reduce or eliminate randomness of their test content.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9173)

### CODE REVIEW

File: scripts/typescript-compatibility-build.yml [hunk 1]
```diff
@@ -155,7 +155,7 @@ jobs:
 
       # Invoke POST API with json body
```
Comment (cliffkoh) [inline scripts/typescript-compatibility-build.yml]: please try to revert any changes to this file

Comment (cliffkoh) [reply scripts/typescript-compatibility-build.yml]: PR #9179 attempts to apply long term fix.

Comment (statm) [reply scripts/typescript-compatibility-build.yml]: welp it looks like this file is screwed even on master (9d0a0b4ba159b96e4dff389c6e51980421e5d006). I'll revert it to previous state.

Comment (ecraig12345) [reply scripts/typescript-compatibility-build.yml]: If you don't mind fixing the indentation, that would be great! It should stick now that Cliff's change is in.

Comment (statm) [reply scripts/typescript-compatibility-build.yml]: @ecraig12345 Just did, waiting for a good CI :)

**Pull request merged**